libs/community/langchain_google_community/vertex_ai_search.py

```diff
@@ -337,10 +386,71 @@ def __init__(self, **kwargs: Any) -> None:
             serving_config=self.serving_config_id,
         )
 
+    def _get_beta_specific_params(self, query: str) -> Dict[str, Any]:
+        """Get parameters that are only available in beta version."""
+
+        params: dict[str, Any] = {}
+
+        if not self.beta:
+            return params
+
+        if (
+            self.custom_embedding is not None
+            or self.custom_embedding_field_path is not None
+        ):
+            if self.custom_embedding is None:
+                raise ValueError(
+                    "Please provide a custom embedding model if you provide a "
+                    "custom_embedding_field_path."
+                )
+            if self.custom_embedding_field_path is None:
+                raise ValueError(
+                    "Please provide a custom_embedding_field_path if you provide a "
+                    "custom embedding model."
+                )
+            if self.custom_embedding_ratio is None:
+                raise ValueError(
+                    "Please provide a custom_embedding_ratio if you provide a "
+                    "custom embedding model or a custom_embedding_field_path."
+                )
+            if not 0 <= self.custom_embedding_ratio <= 1:
```

Comment:
this is not needed, pydantic will handle it because fo your change

Reply:
You are right!